docs/concepts/artifacts.md

```diff
@@ -0,0 +1,228 @@
+# Artifacts
+
+Artifacts offer a means to publish and manage crucial data that is rendered in the Prefect UI, making it easy to see where and when your results were written. Published artifacts may be associated with a particular task run, flow run, or outside a flow run context. Artifacts provide a more advanced way to present information compared to typical logging practices-- including the ability to display reports, tables, markdown, and links to external data.
+## Artifacts Overview <span class="badge experimental"></span>
+
+!!! warning "Artifacts are an experimental feature"
+    Artifact are an experimental feature and are subject to change in future releases.
+
+Whether you're publishing links, markdown, or tables, artifacts provide a powerful and flexible way to showcase important data within your pipeline. By leveraging artifacts, you can easily manage and share critical information with your team, providing valuable insights and context for your workflow runs. With the ability to publish links, markdown, tables, and more, artifacts make it easier than ever to stay organized and informed.
+
+Some potential use cases for artifacts are:
+
+- Debugging: By publishing data that you care about in the UI, you can easily see when and where your results were written. If an artifact doesn't look the way you expect, you can find out which flow run last updated it, and you can click through a link in the artifact to the location where the artifact is stored (such as an S3 bucket).
+- Data quality checks: Artifacts can be used to publish data quality checks from in-progress tasks. This can help ensure that data quality is maintained throughout the pipeline. During long-running tasks such as ML model training, you might use artifacts to publish performance graphs. This can help you visualize how well your models are performing and make adjustments as needed. You can also track the versions of these artifacts over time, making it easier to identify changes in your data.
+- Documentation: Artifacts can be used to publish documentation and sample data from in-progress tasks. This can help you keep track of their work and share information with your colleagues. For instance, artifacts allow you to add a description to let your colleagues know why this piece of data is important. 
+
+## Creating Artifacts
+
+To create artifacts, you need to set `PREFECT_EXPERIMENTAL_ENABLE_ARTIFACTS` to true. Once enabled, you can publish data from task and flow runs or outside of a flow run context. Currently, you can render three artifact types: links, markdown, and tables.
```

Comment:
Do we need this or will we remove the experimental flag before release?

Reply:
```suggestion

```

Reply:
I believe we will be announcing them as experimental? I'm not sure though @billpalombi